### PR TITLE
mixin: add Android Verified Boot support(avb) for cic

### PR DESCRIPTION
**Pull request by GangSecurity**
1. mixin update
2. avb: true

Tracked-On: OAM-90682
Signed-off-by: Chen Gang G <gang.g.chen@intel.com>

### CODE REVIEW

File: cic/addon/pre-requisites/secure [hunk 1]
```diff
@@ -8,6 +8,10 @@ teedata=`sudo parted -l | grep -i teedata`
 trusty_host=`dmesg | grep trusty`
 tos_img=`dmesg | grep trusty | grep 'error -22'`
 trusty_init=`dmesg | grep trusty | grep 'initializ'`
+tos_dev=`sudo blkid|grep "\"tos\"" | awk '{print $1}' | tr -d ':'`
```
Comment (James-Zzz) [inline cic/addon/pre-requisites/secure:11]: suggest to following same coding style, add space before and after '|' symbol:

tos_dev=`sudo blkid | grep "\"tos\"" | awk '{print $1}' | tr -d ':'`
vbmeta_dev=`sudo blkid | grep "\"vbmeta\"" | awk '{print $1}' | tr -d ':'`

File: cic/BoardConfig.mk [hunk 2]
```diff
@@ -20,8 +20,15 @@ BOARD_RECOVERYIMAGE_PARTITION_SIZE ?= 31457280
...
 ##############################################################
+# Source: device/intel/mixins/groups/avb/true/BoardConfig.mk
+##############################################################
+BOARD_AVB_ENABLE := true
+##############################################################
 # Source: device/intel/mixins/groups/boot-arch/project-celadon/BoardConfig.mk
 ##############################################################
+BOARD_BOOTIMAGE_PARTITION_SIZE ?= 31457280
+BOARD_TOSIMAGE_PARTITION_SIZE := 10485760
+
```
Comment (James-Zzz) [inline cic/BoardConfig.mk:31]: remove blank line 31.

File: cic/addon/pre-requisites/secure [hunk 3]
```diff
@@ -26,6 +30,38 @@ if [[ $trusty_host == "" && $security == "true" ]]; then
     echo "security : $security"
 fi
 
+## flash tos and vbmeta to partitions
+if [[ $security == "true" ]]; then
+    if [[ -z $tos_dev || -z $vbmeta_dev ]]; then
+        echo "Partition /tos[10MB] and /vbmeta[1MB] shoud be present"
```
Comment (James-Zzz) [inline cic/addon/pre-requisites/secure:36]: typo: should

File: cic/addon/pre-requisites/secure [hunk 4]
```diff
@@ -26,6 +30,38 @@ if [[ $trusty_host == "" && $security == "true" ]]; then
...
+            echo "error: tos image size is bigger than tos partition size"
+            exit -1
+        fi
+
+        if [[ $vbmeta_img_size -gt `expr $vbmeta_dev_sec_num\*512` ]]; then
+            echo "error: vbmeta image size is bigger than vbmeta partition size"
+            exit -1
+        fi
+
+        tos_blk_count=`expr ${tos_img_size} / 512`
```
Comment (James-Zzz) [inline cic/addon/pre-requisites/secure:53]: do we need to plus one to make sure count would never down round?

tos_blk_count=`expr ${tos_img_size} / 512 + 1`

I checked expr division in shell, expr 15 / 10 = 1, it makes down round.

File: cic/BoardConfig.mk [hunk 5]
```diff
@@ -20,8 +20,15 @@ BOARD_RECOVERYIMAGE_PARTITION_SIZE ?= 31457280
 BOARD_CACHEIMAGE_FILE_SYSTEM_TYPE := ext4
 BOARD_CACHEIMAGE_PARTITION_SIZE ?= 104857600
 ##############################################################
+# Source: device/intel/mixins/groups/avb/true/BoardConfig.mk
+##############################################################
+BOARD_AVB_ENABLE := true
```
Comment (JuhaVainio) [inline cic/BoardConfig.mk:25]: Do we try to keep flags inside Kernelflinger? Is it possible in this case?

File: cic/mixins.spec [hunk 6]
```diff
@@ -26,3 +26,4 @@ camera-ext: ext-camera-only
 bluetooth: cic
 trusty: true(ref_target=cic)
 storage: sdcard-mmc0-usb-sd(adoptablesd=false,adoptableusb=false)
+avb: true
```
Comment (JuhaVainio) [inline cic/mixins.spec:29]: Do we try to keep flags inside Kernelflinger? Is it possible in this case?

Comment (GangSecurity) [reply cic/mixins.spec:29]: No.  The flag is defined by Android, not kernelflinger. It is used to control to generate vbmeta image and sign images by AOSP code.  So should not put into kernelflinger.

File: cic/BoardConfig.mk (hunk 5 above)
Comment (GangSecurity) [reply cic/BoardConfig.mk:25]: No.  The BOARD_AVB_ENABLE flag is defined by Android, not kernelflinger. It is used to control to generate vbmeta image and sign images by AOSP code.  So should not put into kernelflinger.

File: cic/addon/pre-requisites/secure (hunk 4 above)
Comment (GangSecurity) [reply cic/addon/pre-requisites/secure:53]: for vbmeta.img, it is fixed 4096 bytes. tos.img is padding 4096 bytes align. For partitions, always 1M align. Should not need to add round here.

File: cic/BoardConfig.mk (hunk 2 above)
Comment (GangSecurity) [reply cic/BoardConfig.mk:31]: generated by mixin :)

Comment (James-Zzz) [reply cic/BoardConfig.mk:31]: Done. :)

File: cic/addon/pre-requisites/secure (hunk 4 above)
Comment (James-Zzz) [reply cic/addon/pre-requisites/secure:53]: Done. :)

**Pull request merged**